Sources/MoSoCore/MoSoSessionManager.swift

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// TODO: this very simple implementation will change
+public class MoSoSessionManager {
+    public private(set) var user: MoSoUser?
+    public var isLoggedIn: Bool {
+        user != nil
+    }
+
+    public init(user: MoSoUser? = nil) {
+        self.user = user
+    }
+
+    public func loggedIn(_ user: MoSoUser) {
+        self.user = user
+    }
+
+    public func loggedOut() {
+        user = nil
+    }
+
+    public func activeSession() throws -> MoSoSession {
```

Comment:
In general: I like throwing because returning nil often hides an error. Throws make it very explicit what is happening.

In this case we may want to have different flows based on if the user needs to refresh their token (which we may be able to do automatically) and there is no user logged in, which will require us to launch the login flow.